### ISSUE TITLE
`parse_part` dropping any instance of `\r\n`

### ISSUE THREAD

**Issue by sgrove** [cryptosense/multipart-form-data#17]
Took awhile to track this down, but our binary uploads may contain `\r\n` in the file bodies somewhere, and it looks like Multipart is stripping every instance, leading to invalid files being stored to disk.

It looks like it might be here https://github.com/cryptosense/multipart-form-data/blob/master/lib/multipart.ml#L163 where all of the lines, having been split on `\r\n`, are simply concatenated together as-is, not restoring the `\r\n` that was there beforehand (to be clear, we need the bytes `0d0a` to be in the final filestream).

I tried a few different hacks, but I'm not very familiar with `Lwt_stream`, and couldn't quite figure out how best to get it to interpose the streams I need. Any hints on how to make it work?

**Comment by wokalski** [comment]
@sgrove it seems like you handled this and other issues in your fork. Maybe you could PR it if it's been working for you so far?

**Comment by emillon** [comment]
Right, sorry for not answering before. As mentioned in #16 I'm not super convinced about the current API, and this package will probably go through significant changes, but that shouldn't prevent it from being correct :) If you have a working patch I'm happy to merge it. Thanks!

**Comment by xguerin** [comment]
This has become very relevant for me. I'll see if I can address that issue.

**Comment by emillon** [comment]
I'm not a maintainer anymore, but if this package does not work, you might want to try https://github.com/dinosaure/multipart_form (not released yet).

**Comment by xguerin** [comment]
Thanks. I decided for a direct API call instead, but I'll take a look if I decide to use `multipart` in the future.

**Comment by lindig** [comment]
I also suspect that handling of `\r\n` is not correct: https://github.com/rgrinberg/opium/issues/258#issue-792867630
